neuralprophet/conformal.py

```diff
@@ -190,3 +190,62 @@ def plot(self, plotting_backend: str):
             )
         if plotting_backend in ["matplotlib", "plotly"] and matplotlib.is_interactive():
             fig.show()
+
+    def evaluate(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Evaluate conformal prediction on test dataframe.
+
+        Parameters
+        ----------
+            df : pd.DataFrame
+                test dataframe with the conformal prediction intervals
+
+        Returns
+        -------
+            pd.DataFrame
+                table containing evaluation metrics such as interval_width and miscoverage_rate
+        """
+        eval_df = pd.DataFrame()
+        for step_number in range(1, self.n_forecasts + 1):
+            q_hat = self.q_hats[step_number - 1]
+            if self.method == "naive":
+                # Interval width (efficiency metric)
+                interval_width = q_hat * 2
+                # Miscoverage rate (validity metric)
+                n_covered = df.apply(
+                    lambda row: bool(
+                        row[f"yhat{step_number} - qhat{step_number}"]
+                        <= row["y"]
+                        <= row[f"yhat{step_number} + qhat{step_number}"]
+                    ),
+                    axis=1,
+                )
+                coverage_rate = n_covered.sum() / len(df)
+                miscoverage_rate = 1 - coverage_rate
+            elif self.method == "cqr":
+                quantile_hi = str(max(quantiles) * 100)
+                quantile_lo = str(min(quantiles) * 100)
```

Comment:
Did you mean self.quantiles here instead of quantiles?

Reply:
I fixed, great catch!